sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+            self._variable = variable
+        else:
+            raise TypeError("""The variable should be a Symbol object.""")
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+        self._load = 0
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def length(self):
+        """Length of the Beam."""
+        return self._length
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def variable(self):
+        return self._variable
+
+    @variable.setter
+    def variable(self, v):
+        if isinstance(v, Symbol):
+            self._variable = v
+        else:
+            raise TypeError("""The variable should be a Symbol object.""")
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        The dictionary has three kewwords namely moment, slope and deflection.
+        The value of each keyword is a list of tuple, where each tuple
+        contains loaction and value of a boundary condition in the format
+        (location, value).
+
+        Examples
+        ========
+        There is a beam of length 4 meters. The bending moment at 0 should be 4
+        and at 4 it should be 0. The slope of the beam should be 1 at 0. The
+        deflection should be 2 at 0.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.bc_moment = [(0, 4), (4, 0)]
+        >>> b.bc_deflection = [(0, 2)]
+        >>> b.bc_slope = [(0, 1)]
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+
+        Here the deflection of the beam should be ``2`` at ``0``.
+        Similarly, the slope of the beam should be ``1`` at ``0`` and
+        there are two boundary conditions for bending moment. The bending
+        moment of the beam should be ``4`` at ``0`` and ``0`` at ``4``
+        """
+        return self._boundary_conditions
+
+    @property
+    def bc_moment(self):
+        return self._boundary_conditions['moment']
+
+    @bc_moment.setter
+    def bc_moment(self, m_bcs):
+        self._boundary_conditions['moment'] = m_bcs
```

Comment:
Does `b.bc_moment.append((1, 2))` work? Can I use the attribute as a list with all of list's methods and such? Please add tests showing this if you haven't.